HeterogeneousCore/CUDAUtilities/src/CachingDeviceAllocator.h

```diff
@@ -545,6 +546,7 @@ namespace notcub {
     cudaError_t DeviceFree(int device, void *d_ptr) {
       int entrypoint_device = INVALID_DEVICE_ORDINAL;
       cudaError_t error = cudaSuccess;
+      std::unique_lock<std::mutex> mutex_locker(mutex, std::defer_lock);
```

Comment:
Please add a comment along
```suggestion
      // CMS: use RAII instead of (un)locking explicitly
      std::unique_lock<std::mutex> mutex_locker(mutex, std::defer_lock);
```